tests/agent/test_session_persister.py

```
@@ -0,0 +1,720 @@
+class TestCallerSitesUsePublicAPI:
+    """Verify that callers use the public API, not private internals.
+
+    Uses AST inspection to check that:
+    - batch_runner.py does NOT call agent._convert_to_trajectory_format (crash bug)
+    - run_agent.py does NOT call _persister._convert_to_trajectory_format (private coupling)
+    """
+
+    def test_batch_runner_no_agent_convert_to_trajectory(self):
+        """batch_runner.py must NOT call agent._convert_to_trajectory_format (crashes post-refactor)."""
+        import ast
+        from pathlib import Path
```

Ruff (0.15.2)

[error] Redefinition of unused `Path` from line 9: `Path` redefined here

Remove definition: `Path`

(F811)